cli/connector/eval_plain_text.go

```diff
@@ -0,0 +1,389 @@
+}
+
+func readFromPlainTextConnLua(conn net.Conn, opts EvalPlainTextOpts) ([]byte, error) {
+	return readFromPlainTextConn(conn, endOfLuaOutput, opts)
+}
+
+func processEvalTarantoolResYAML(resBytes []byte, result interface{}) ([]interface{}, error) {
+	evalResults := []PlainTextEvalRes{}
+	if err := yaml.UnmarshalStrict(resBytes, &evalResults); err != nil {
+		errorStrings := make([]map[string]string, 0)
+		if err := yaml.UnmarshalStrict(resBytes, &errorStrings); err == nil {
+			if len(errorStrings) > 0 {
+				err, found := errorStrings[0]["error"]
+				if found {
+					return nil, fmt.Errorf("Syntax error: %s", err)
+				}
+			}
+
+		}
+
+		return nil, fmt.Errorf("Failed to parse eval result: %s", err)
+	}
+
+	if len(evalResults) != 1 {
+		return nil, fmt.Errorf("Expected one result, found %d", len(evalResults))
+	}
+
+	evalResult := evalResults[0]
+
+	if !evalResult.Success {
+		return nil, fmt.Errorf(evalResult.ErrStr)
+	}
+
+	if result != nil {
+		if err := yaml.UnmarshalStrict([]byte(evalResult.DataEnc), result); err != nil {
+			return nil, fmt.Errorf("Failed to parse eval result: %s", err)
+		}
+
+		return nil, nil
+	}
+
+	var data []interface{}
+	if err := yaml.UnmarshalStrict([]byte(evalResult.DataEnc), &data); err != nil {
+		return nil, fmt.Errorf("Failed to parse eval result: %s", err)
+	}
+
+	return data, nil
+}
+
+// func processEvalTarantoolResLua(resBytes []byte) (*PlainTextEvalRes, error) {
+// 	L := lua.NewState()
+// 	defer L.Close()
+
+// 	doString := fmt.Sprintf(`res = %s`, resBytes)
+
+// 	if err := L.DoString(doString); err != nil {
+// 		return nil, err
+// 	}
+
+// 	luaRes := L.Env.RawGetString("res")
+
+// 	if luaRes.Type() == lua.LTString {
+// 		return nil, fmt.Errorf("Syntax error: %s", lua.LVAsString(luaRes))
+// 	}
+
+// 	successLV := L.GetTable(luaRes, lua.LString("success"))
+// 	messageLV := L.GetTable(luaRes, lua.LString("err"))
+// 	// I've no idea how to get interface{} value from a map =(
+// 	encodedDataLV := L.GetTable(luaRes, lua.LString("data"))
+
+// 	success := lua.LVAsBool(successLV)
+// 	message := lua.LVAsString(messageLV)
+// 	encodedData := lua.LVAsString(encodedDataLV)
+
+// 	res := PlainTextEvalRes{
+// 		Success: success,
+// 		ErrStr:  message,
+// 	}
+
+// 	if err := yaml.Unmarshal([]byte(encodedData), &res.Data); err != nil {
+// 		return nil, fmt.Errorf("Failed to unmarshal data: %s", err)
+// 	}
+
+// 	return &res, nil
+// }
+
+var (
+	callFuncTmpl = `return {{ .FunctionName }}(...)`
+)
+
+const (
+	evalFuncYAMLTmpl = `
+local function __func(...)
+	require('fiber').self().storage.console = nil
+	{{ .FunctionBody }}
+end
+
+local ok, ret = pcall(function()
+	local args = require('msgpack').decode(string.fromhex('{{ .ArgsEncoded }}'))
+	local ret = { __func(unpack(args)) }
+	return ret
+end)
+
+if ret == nil then ret = box.NULL end
+
+if not ok then
+	return { success = false, err = ret}
+end
+
+return { success = true, data_enc = require('yaml').encode(ret) }
```

Comment:
Why can't you use something like:

```
return {message = require('yaml').encode(ret)} -- for success
return {err = err} -- for errors
```

Reply:
I just think that `res.Success` check looks good and it's pretty clear, but it also can be `res.Err != nil` check.